src/main/java/core/basesyntax/serviceimpl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.service.ReaderService;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReaderServiceImpl implements ReaderService<List<String>> {
+    private final List<String> lines = new ArrayList<>();
```

Comment:
The 'lines' list should not be a class member since it's only used in the 'readFromFile' method. This could lead to unwanted state retention between method calls. Declare it inside the method instead.